src/coreclr/vm/jitinterface.cpp

```diff
@@ -1653,60 +1772,55 @@ void CEEInfo::getThreadLocalStaticBlocksInfo (CORINFO_THREAD_STATIC_BLOCKS_INFO*
         MODE_PREEMPTIVE;
     } CONTRACTL_END;
 
-    JIT_TO_EE_TRANSITION_LEAF();
+    JIT_TO_EE_TRANSITION();
 
+    size_t threadStaticBaseOffset = 0;
+
+#ifdef _MSC_VER
     pInfo->tlsIndex.addr = (void*)static_cast<uintptr_t>(_tls_index);
     pInfo->tlsIndex.accessType = IAT_VALUE;
 
     pInfo->offsetOfThreadLocalStoragePointer = offsetof(_TEB, ThreadLocalStoragePointer);
-    if (isGCType)
-    {
-        pInfo->offsetOfThreadStaticBlocks = ThreadLocalOffset(&t_GCThreadStaticBlocks);
-        pInfo->offsetOfMaxThreadStaticBlocks = ThreadLocalOffset(&t_GCMaxThreadStaticBlocks);
-    }
-    else
-    {
-        pInfo->offsetOfThreadStaticBlocks = ThreadLocalOffset(&t_NonGCThreadStaticBlocks);
-        pInfo->offsetOfMaxThreadStaticBlocks = ThreadLocalOffset(&t_NonGCMaxThreadStaticBlocks);
-    }
+    threadStaticBaseOffset = ThreadLocalOffset(&t_ThreadStatics);
 
-    pInfo->offsetOfGCDataPointer = static_cast<uint32_t>(PtrArray::GetDataOffset());
+#elif defined(TARGET_OSX)
 
-    JIT_TO_EE_TRANSITION_LEAF();
-}
-#else
+    // For OSX x64/arm64, need to get the address of relevant tlv_get_addr of thread static
+    // variable that will be invoked during runtime to get the right address of corresponding
+    // thread.
+    pInfo->descrAddrOfMaxThreadStaticBlock = (size_t)getThreadStaticsBaseOffset();
```

Comment:
This can use consistent naming. It is called `descrAddrOfMaxThreadStaticBlock` in some places, and `ThreadStaticsBaseOffset` in others.